crates/solver/src/driver.rs

```diff
@@ -563,10 +563,14 @@ impl Driver {
     }
 }
 
-fn is_only_selling_trusted_tokens(settlement: &Settlement, token_list: &TokenList) -> bool {
-    !settlement
-        .traded_orders()
-        .any(|order| token_list.get(&order.creation.sell_token).is_none())
+fn is_only_selling_and_buying_trusted_tokens(
+    settlement: &Settlement,
+    token_list: &TokenList,
+) -> bool {
+    !settlement.traded_orders().any(|order| {
```

Comment:
nit: Seems more intuitive to use to use [`Iterator::all()`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.all) to avoid a double negative here.